File: 287_FindTheDuplicateNumber.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        mySet = set()
        
        for element in nums:
            if element in mySet:
                return element
            mySet.add(element)
        
        

#Solution 2
        nums.sort()
        #Constant Extra Space
        for i in range(1,len(nums)):
            if nums[i] == nums[i-1]:
                return nums[i]

File: 287_FindTheDuplicateNumber.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        map<int,int> mp;
        int retVal = 0;
        
        for(int i = 0; i < nums.size(); i++)
        {
            mp[nums[i]] += 1;
            if(mp[nums[i]] == 2)
            {
                retVal = nums[i];
            }
        }
        
        return retVal;
        
    }
};